File: C_CPP/baskara.c
#include <stdio.h>
#include <math.h>

//programa para resolver as raízes de uma equação do II grau

int main(void){
     
    double a,b,c; //coeficientes
    double x1,x2; //raízes
    double delta; //discriminante
    //char resp;
    
    printf("Walescko Software Solutions\n");
    printf("Raízes da Equação de II Grau\n");
    printf("Coeficiente a: ");
    scanf("%lf", &a);
    printf("Coeficiente b: ");
    scanf("%lf", &b);
    printf("Coeficiente c: ");
    scanf("%lf", &c);
    
    delta = b*b - 4*a*c;
    
    printf("valor do discriminante %f\n", delta);

    if ((a == 0 ) || (delta < 0)) {
        printf("A equação não possui raízes reais\n");
    } else if (delta == 0){
        x1 = -b/(2*a);
        printf("A equação possui duas raízes reais e iguais x1 = x2 = %f\n", x1);
        } else if (delta > 0){
            x1 = (-b + sqrt(delta))/(2*a);
            x2 = (-b - sqrt(delta))/(2*a);
            printf("A equação possui duas raízes reais iguais x1 = %f e x2 = %f\n", x1, x2);
        }
        
    return 0;
}

File: java/baskara.java
import java.io.*;
import java.text.DecimalFormat;
import java.util.Scanner;

class baskara{
    public static void main(String args[]){
        
        double a,b,c;
        double delta;
        double x1, x2;
        Scanner s = new Scanner(System.in);
        DecimalFormat df = new DecimalFormat("00.00");
        
        System.out.println();
        System.out.println("WADAJU SOFTWARE SOLUTIONS");
        System.out.println("Equação do II Grau - Raízes");
        System.out.println();
        System.out.println("Entre com o valor dos coeficientes da Equação Quadrática:");
        System.out.print("a = ");
        a = s.nextDouble();
        System.out.print("b = ");
        b = s.nextDouble();
        System.out.print("c = ");
        c = s.nextDouble();
        
        delta = b*b - 4 * a * c;
        System.out.println("Delta = " + df.format(delta));
        System.out.println();
        
        //raízes da equacao
        
        
        if ((a == 0) || (delta < 0 )){
            System.out.println("Não tem raízes reais!");
            } else {
            
        if (delta == 0 ){
            x1 = -b/(2*a);
            System.out.println("A equação tem duas raízes reais idênticas X1 = X2 = "+df.format(x1));
            } else {
            x1 = (-b + Math.sqrt(delta))/(2*a);
            x2 = (-b - Math.sqrt(delta))/(2*a);
            System.out.println("A equação tem duas raízes reais distintas X1 = "+df.format(x1)+" e X2 = "+df.format(x2));
            }
            }
            
            
        // vértice da parábola
            double xv, yv;
            xv = (-b)/(2*a);
            yv = -delta/(4*a);
        //       if ( a < 0 ) {
            System.out.println("O vértice da parabóla é Xv = "+df.format(xv)+ " e Yv = "+df.format(yv));
      //      } else {
    //                 System.out.println("O vértice da parabóla é Xv = "+df.format(xv)+ " e Yv = "+df.format(yv));
    
            System.out.println();
            System.out.println("fim");
 }
}
